src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurDetailLineGroupForCollector.java

```diff
@@ -79,47 +142,47 @@ protected String defaultToDashesIfBlank(String value, String propertyName) {
         return StringUtils.defaultIfBlank(value, collectorHelper.getDashOnlyPropertyValue(propertyName));
     }
 
-    public List<OriginEntryFull> buildOriginEntries() {
-        List<OriginEntryFull> originEntries = new ArrayList<>();
+    /**
+     * Generates OriginEntryFull BOs from the SAE lines and passes them to the given Consumer.
+     * If there was an error condition that should cause all of the lines of the current Report ID
+     * to be skipped (such as a Cash Advance line missing a corresponding Requested Cash Advance BO),
+     * the SAE lines and their error messages will be passed to the given BiConsumer instead.
+     * 
+     * @param entryConsumer A Consumer that the generated OriginEntryFull BOs can be passed to.
+     * @param lineErrorReporter A BiConsumer that can accept an SAE line and its error message.
+     */
+    public void buildAndAddOriginEntries(Consumer<OriginEntryFull> entryConsumer,
+            BiConsumer<ConcurStandardAccountingExtractDetailLine, String> lineErrorReporter) {

```

Comment:
This was meant to simplify calling the ConcurStandardAccountingExtractCollectorBatchBuilder.reportUnprocessedLine() method, without having to pass a reference to the ConcurStandardAccountingExtractCollectorBatchBuilder that is calling this buildAndAddOriginEntries() method.